packages/runtime/container-runtime/src/test/runningSummarizer.spec.ts

```diff
@@ -647,6 +670,69 @@ describe("Runtime", () => {
                     ]), "unexpected log sequence 3");
                 });
             });
+
+            describe("Disabled Heuristics", () => {
+                it("Should not summarize after time or ops", async () => {
+                    await startRunningSummarizer({ disableHeuristics: true });
+
+                    await emitNextOp(summaryConfig.maxOps + 1);
+                    assertRunCounts(0, 0, 0, "should not summarize after maxOps");
+
+                    await tickAndFlushPromises(summaryConfig.idleTime + 1);
+                    assertRunCounts(0, 0, 0, "should not summarize after idleTime");
+
+                    await tickAndFlushPromises(summaryConfig.maxTime + 1);
+                    assertRunCounts(0, 0, 0, "should not summarize after maxTime");
+
+                    await emitNextOp(summaryConfig.maxOps * 3 + 10000);
+                    await tickAndFlushPromises(summaryConfig.maxTime * 3 + summaryConfig.idleTime * 3 + 10000);
+                    assertRunCounts(0, 0, 0, "make extra sure");
+                });
+
+                it("Should not summarize before closing", async () => {
+                    await startRunningSummarizer({ disableHeuristics: true });
+
+                    await emitNextOp(51); // hard-coded to 50 for now
+                    const stopP = summarizer.waitStop();
+                    await flushPromises();
+                    await emitAck();
+                    await stopP;
+
+                    assertRunCounts(0, 0, 0, "should not perform lastSummary");
+                });
+
+                it("Should not summarize immediately if summary ack is missing at startup when disabled", async () => {
+                    assertRunCounts(0, 0, 0);
+                    // Simulate as summary op was in op stream.
+                    const summaryTimestamp = Date.now();
+                    emitBroadcast(summaryTimestamp);
+
+                    let startStatus: "starting" | "started" | "failed" = "starting";
+                    startRunningSummarizer({ disableHeuristics: true }).then(
+                        () => startStatus = "started", () => startStatus = "failed");
+                    await flushPromises();
+                    assert.strictEqual(startStatus, "starting",
+                        "RunningSummarizer should still be starting since outstanding summary op");
```

Comment:
I should have looked at this message here and things may have been clearer :-). Can you add this as a comment above the assert too?